rules/falco_rules.yaml

```diff
@@ -2210,6 +2210,27 @@
   priority: NOTICE
   tags: [network, aws, container, mitre_discovery]
 
+# In a local/user rules file, you could override this macro to
+# explicitly enumerate the container images that you want to allow
+# access to GCE metadata. In this main falco rules file, there isn't
+# any way to know all the containers that should have access, so any
+# container is alllowed, by repeating the "container" macro. In the
+# overridden macro, the condition would look something like
+# (container.image.repository = vendor/container-1 or
+# container.image.repository = vendor/container-2 or ...)
+- macro: gce_metadata_containers
+  condition: container
```

Comment:
This macro has two responsibilities:
1. Enabling the rule
2. Filtering for container images

I think we would rather want two different macros: 
- one that does the enable/disable using the never_true condition [example](https://github.com/falcosecurity/falco/blob/dev/rules/falco_rules.yaml#L459) - this one will be used at the very start of the rule - in this way we disable without evaluating anything else 
- one to do the filtering (we can keep the one you already wrote, however I would put one of the examples you wrote in it instead of `container`

Then we document the rule by saying, if you want to enable this one you overwrite to `always_true` the first macro and then you overwrite the second macro with the images you want to filter on.

Wdyt ?

Thanks for the good work again, we are getting closer! I would really love to have this merged

Reply:
Thank you for your review.
I thought your suggestion made my PR much better.

> 1. Enabling the rule
- I added "consider_gce_metadata_access" macro as flag of this rule.

> 2. Filtering for container images
- We can have default configuration to "gce_metadata_containers" now because we have "consider_gce_metadata_access".
  - so I set "(k8s.ns.name = "kube-system")" as default condition.
  - (I tried this rule on GKE, and there is no false positive)

- and I rewrote comments to simplify.

Reply:
We need different pre-configuration now.

### Configuration
- Overwrite consider_gce_metadata_access
```
- macro: consider_gce_metadata_access
  condition: (always_true)
```